### PR TITLE
using /v1alpha2/jenkins replace /jenkins.kubesphere.io api

### PR DESCRIPTION
**Pull request by shaowenchen**
**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:

There are some APIs forwarding by caddy in 2.1.1.  Now，we can use /kapis/devops.kubesphere.io/v1alpha2/jenkins to replace it.

### CODE REVIEW

File: pkg/kapis/devops/v1alpha2/handler.go [hunk 1]
```diff
@@ -34,6 +40,17 @@ type PipelineSonarHandler struct {
 	pipelineSonarGetter devops.PipelineSonarGetter
 }
 
+func jenkinsProxy(request *restful.Request, response *restful.Response) {
+	u := request.Request.URL
+	u.Host = "allinone.ks.dev.chenshaowen.com:30180"
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha2/handler.go:45]: What's this? why the host has to be this ?

**Comment by zryfish** [comment]
Why are we back to old days, proxying requests directly to Jenkins? I think we've already decoupled from Jenkins.

File: pkg/kapis/devops/v1alpha2/handler.go (hunk 1 above)
Comment (shaowenchen) [reply pkg/kapis/devops/v1alpha2/handler.go]: It's testing code. Removed.

**Comment by shaowenchen** [comment]
> Why are we back to old days, proxying requests directly to Jenkins? I think we've already decoupled from Jenkins.

There are some APIs that get files from Jenkins，e.g download artifacts or test reports. These interfaces have no definite format but the same prefix. 

In 2.1, we proxy it using caddy in the API gateway. @zryfish

**Comment by zryfish** [comment]
> > Why are we back to old days, proxying requests directly to Jenkins? I think we've already decoupled from Jenkins.
> 
> There are some APIs that get files from Jenkins，e.g download artifacts or test reports. These interfaces have no definite format but the same prefix.
> 
> In 2.1, we proxy it using caddy in the API gateway. @zryfish

Which APIs needed to proxy to Jenkins? What I really want to say is we shouldn't use Jenkins API directly, it's tightly coupled.

File: pkg/kapis/devops/v1alpha2/register.go [hunk 2]
```diff
@@ -59,6 +64,11 @@ func AddToContainer(container *restful.Container, ksInformers externalversions.S
 		return err
 	}
 
+	err = AddJenkinsToContainer(ws, endpoint)
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha2/register.go:67]: If endpoint is empty, means Devops is not enabled, `AddJenkinsToContainer` will return an error

Comment (shaowenchen) [reply pkg/kapis/devops/v1alpha2/register.go]: fixed

**Comment by shaowenchen** [comment]
> > > Why are we back to old days, proxying requests directly to Jenkins? I think we've already decoupled from Jenkins.
> > 
> > 
> > There are some APIs that get files from Jenkins，e.g download artifacts or test reports. These interfaces have no definite format but the same prefix.
> > In 2.1, we proxy it using caddy in the API gateway. @zryfish
> 
> Which APIs needed to proxy to Jenkins? What I really want to say is we shouldn't use Jenkins API directly, it's tightly coupled.

/job/new-devopsfvs8z/job/archive-zip/job/1/artifact/a/b/c/newfile.txt

If it's tightly coupled, we can add prefix /job/new-devopsfvs8z/job/archive-zip/job/1/artifact to check.

**Comment by zryfish** [comment]
> > > > Why are we back to old days, proxying requests directly to Jenkins? I think we've already decoupled from Jenkins.
> > > 
> > > 
> > > There are some APIs that get files from Jenkins，e.g download artifacts or test reports. These interfaces have no definite format but the same prefix.
> > > In 2.1, we proxy it using caddy in the API gateway. @zryfish
> > 
> > 
> > Which APIs needed to proxy to Jenkins? What I really want to say is we shouldn't use Jenkins API directly, it's tightly coupled.
> 
> /job/new-devopsfvs8z/job/archive-zip/job/1/artifact/a/b/c/newfile.txt
> 
> If it's tightly coupled, we can add prefix /job/new-devopsfvs8z/job/archive-zip/job/1/artifact to check.

It's our job

File: pkg/kapis/devops/v1alpha2/register.go [hunk 3]
```diff
@@ -685,3 +695,25 @@ func AddS2IToWebService(webservice *restful.WebService, ksClient versioned.Inter
...
+	}
+	parse.Path = strings.Trim(parse.Path, "/")
+	webservice.Route(webservice.GET("/jenkins/{path:*}").
+		To(func(request *restful.Request, response *restful.Response) {
+			u := request.Request.URL
+			u.Host = parse.Host
+			u.Scheme = parse.Scheme
+			jenkins.SetBasicBearTokenHeader(&request.Request.Header)
+			u.Path = strings.Replace(request.Request.URL.Path, fmt.Sprintf("/kapis/%s/%s/jenkins", GroupVersion.Group, GroupVersion.Version), "", 1)
+			httpProxy := proxy.NewUpgradeAwareHandler(u, http.DefaultTransport, false, false, nil)
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha2/register.go:715]: ErrorResponser can be useful for debugging, please don't pass nil

Comment (shaowenchen) [reply pkg/kapis/devops/v1alpha2/register.go]: fixed

**Pull request merged**